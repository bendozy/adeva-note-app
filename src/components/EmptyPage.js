import React from 'react';
import PropTypes from 'prop-types';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(theme => ({
  body: {
    marginTop: theme.spacing(5),
  },
}));

const EmptyPage = ({ title, text }) => {
  const classes = useStyles();

  return (
    <Grid>
      <Grid>
        <Typography variant="h4" color="inherit" noWrap>
          {title}
        </Typography>
      </Grid>
      <Grid>
        <Typography
          variant="body1"
          component="p"
          color="inherit"
          className={classes.body}
        >
          {text}
        </Typography>
      </Grid>
    </Grid>
  );
};

EmptyPage.defaultProps = {
  title: 'Page Not Found',
  text: 'The requested url could not be found',
};

EmptyPage.propTypes = {
  title: PropTypes.string,
  text: PropTypes.string,
};

export default EmptyPage;
