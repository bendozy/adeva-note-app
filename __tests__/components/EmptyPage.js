import React from 'react';
import toJson from 'enzyme-to-json';
import Typography from '@material-ui/core/Typography';
import EmptyPage from '../../src/components/EmptyPage';

describe('<EmptyPage />', () => {
  it('should render a <EmptyPage /> component with default props', () => {
    const wrapper = shallow(<EmptyPage />);

    expect(wrapper.find(Typography)).toHaveLength(2);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

