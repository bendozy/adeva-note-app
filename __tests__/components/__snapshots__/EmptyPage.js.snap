// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyPage /> should render a <EmptyPage /> component with default props 1`] = `
<WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Typography))
      color="inherit"
      noWrap={true}
      variant="h4"
    >
      Page Not Found
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-body-1"
      color="inherit"
      component="p"
      variant="body1"
    >
      The requested url could not be found
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
